snippets/csharp/System.Collections.Generic/IEqualityComparerT/Overview/program.cs

```diff
@@ -2,81 +2,69 @@
 using System;
 using System.Collections.Generic;
 
-class Example
+static class Example
 {
-   static void Main()
-   {
-      BoxEqualityComparer boxEqC = new BoxEqualityComparer();
-
-      var boxes = new Dictionary<Box, string>(boxEqC);
-
-      var redBox = new Box(4, 3, 4);
-      AddBox(boxes, redBox, "red");
+    static void Main()
+    {
+        BoxEqualityComparer comparer = new();
 
-      var blueBox = new Box(4, 3, 4);
-      AddBox(boxes, blueBox, "blue");
+        Dictionary<Box, string> boxes = new(comparer);
 
-      var greenBox = new Box(3, 4, 3);
-      AddBox(boxes, greenBox, "green");
-      Console.WriteLine();
+        AddBox(new Box(4, 3, 4), "red");
+        AddBox(new Box(4, 3, 4), "blue");
+        AddBox(new Box(3, 4, 3), "green");
 
-      Console.WriteLine("The dictionary contains {0} Box objects.",
-                        boxes.Count);
-   }
+        Console.WriteLine($"The dictionary contains {boxes.Count} Box objects.");
 
-   private static void AddBox(Dictionary<Box, String> dict, Box box, String name)
-   {
-      try {
-         dict.Add(box, name);
-      }
-      catch (ArgumentException e) {
-         Console.WriteLine("Unable to add {0}: {1}", box, e.Message);
-      }
-   }
+        void AddBox(Box box, string name)
```

Comment:
Is this helper method and try/catch block strictly necessary for this example?

Reply:
I think it's trying to demonstrate that the comparer is working, by disallowing two values that compare to equal to be added to the collection. The same thing could be demonstrated more simply by using `ContainsKey` I suppose.

The whole example might be simpler yet if it used a `HashSet<>`.

Reply:
I was suggesting that the throwing behavior might be sufficient in demonstrating this, but if this is more consistent with idioms followed in the docs I'm fine with that.

Reply:
This is one area where our style guide allows flexibility. The throwing behavior would be correct in many real-world scenarios, so that's fine. The counter is that explicitly catching and noting the exception thrown provides "documentation" of sorts for exceptions readers may encounter. It's fine as written.